application/src/main/java/org/thingsboard/server/service/ws/DefaultWebSocketService.java

```diff
@@ -843,6 +851,17 @@ private void sendPing() {
                 }));
     }
 
+    private void cleanupClosedSessions() {
+        wsSessionsMap.keySet().forEach(sessionId -> {
+            if (!msgEndpoint.isOpen(sessionId)) {
+                log.debug("[{}] Session does not exists, removing all subscriptions", sessionId);
+                wsSessionsMap.remove(sessionId);
```

Comment:
We are lucky that ConcurrentHashMap does not throw ConcurrentModificationException :)